javaslang/src/test/java/javaslang/collection/PriorityQueueTest.java

```diff
@@ -216,15 +216,12 @@ public void shouldFailDequeueOfEmpty() {
         empty().dequeue();
     }
 
-    // -- toSortedQueue
+    // -- toPriorityQueue
 
     @Test
-    public void shouldKeepInstanceOfSortedQueue() {
-        final SerializableComparator<Integer> comparator = naturalComparator();
-        final PriorityQueue<Integer> queue = PriorityQueue.of(comparator, 1, 3, 2);
```

Comment:
Comparators (or functions in general) cannot be compared for equality. Therefore `PriorityQueue.of(comparator, ...)` always has to return a new instance.